packages/true-time/src/index.ts

```diff
@@ -27,26 +40,42 @@ export class TrueTime {
     );
   }
 
+  async setServerTime() {
+    try {
+      const response = await fetch(this.#url);
+
+      if (response.ok && response.headers.has('date')) {
+        this.#serverTime = new Date(response.headers.get('date')!).getTime();
+        // eslint-disable-next-line no-restricted-syntax
+        this.#synced = Date.now();
+      }
+    } catch (error) {
+      console.error(error);
+    }
+  }
+
   /**
    * Use this method to synchronize time with the server.
    *
    * @param url - server url
    */
   async synchronize() {
-    if (this.#synced) {
+    // Synchronize at max once every 1 000 000 miliseconds
```

Comment:
If this should be allowed to run more than once, I can't see how it can work if not also setting `clientStartTime`?